app.js

```diff
@@ -26,10 +66,17 @@ app.use(function(req, res, next) {
 });
 
 app.get('/validate', function (req, res, next) {
-  const token = firebase.auth().createCustomToken(req.query.token);
-  res.json({ token });
+  var apiToken = req.query.token;
+  if (isValidToken(apiToken)) {
+    getApiUser(apiToken).then(function() {
+      const token = firebase.auth().createCustomToken(req.query.uid);
+      res.json({ token });
+    })
+  } else {
+    console.log('Invalid token')
```

Comment:
You should probably respond with a HTTP 422 or something here that is useful for the client making the call to understand why it didn't work.